great_expectations/datasource/data_connector/configured_asset_gcs_data_connector.py

```diff
@@ -1,7 +1,171 @@
+
+    As much of the interaction with the SDK is done through a GCS Storage Client, please refer to the official
+    docs if a greater understanding of the supported authentication methods and general functionality is desired.
+    Source: https://googleapis.dev/python/google-api-core/latest/auth.html
+    """
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        bucket_or_name: str,
+        assets: dict,
+        execution_engine: Optional[ExecutionEngine] = None,
+        default_regex: Optional[dict] = None,
+        sorters: Optional[list] = None,
+        prefix: Optional[str] = None,
+        delimiter: Optional[str] = None,
+        max_results: Optional[int] = None,
+        gcs_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        """
+        ConfiguredAssetDataConnector for connecting to GCS.
+
+        Args:
+            name (str): required name for DataConnector
+            datasource_name (str): required name for datasource
+            container (str): container name for Google Cloud Storage
+            assets (dict): dict of asset configuration (required for ConfiguredAssetDataConnector)
+            execution_engine (ExecutionEngine): optional reference to ExecutionEngine
+            default_regex (dict): optional regex configuration for filtering data_references
+            sorters (list): optional list of sorters for sorting data_references
+            prefix (str): GCS prefix
+            delimiter (str): GCS delimiter
+            max_results (int): max blob filepaths to return
+            gcs_options (dict): wrapper object for optional GCS **kwargs
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.debug(f'Constructing ConfiguredAssetGCSDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            assets=assets,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+        self._bucket_or_name = bucket_or_name
+        self._prefix = prefix
+        self._delimiter = delimiter
+        self._max_results = max_results
+
+        if gcs_options is None:
+            gcs_options = {}
+
+        try:
+            credentials = None  # If configured with gcloud CLI / env vars
+            if "filename" in gcs_options:
+                credentials = service_account.Credentials.from_service_account_file(
+                    **gcs_options
+                )
+            elif "info" in gcs_options:
+                credentials = service_account.Credentials.from_service_account_info(
+                    **gcs_options
+                )
```

Comment:
Schema validation will ensure that we only have one of these options (can also have neither).